### DISCUSSION TITLE
Valetudo 2021.09.1

### DISCUSSION THREAD

**Discussion by Hypfer** [Hypfer/Valetudo#1098] [Release Discussions]
<div align="center">
    <img src="https://github.com/Hypfer/Valetudo/blob/master/assets/logo/valetudo_logo_with_name.svg" width="800" alt="valetudo">
    <p align="center"><h2>2021.09.1</h2></p>
</div>

This release is packed with UI improvements thanks to @ccoors

## Event UI

The new UI now subscribes to the ValetudoEvents endpoint and allows for interaction with them.

![image](https://user-images.githubusercontent.com/974410/133927122-054ed31b-7a47-447f-9193-6e92ca64757b.png)

This is helpful as they might contain errors that you've missed.
Also, your robot might require confirmation from you to store a new map so make sure to keep an eye on that.

![image](https://user-images.githubusercontent.com/974410/133927125-d069022e-8a18-4bd3-b879-734a4f9aba96.png)

## Timer UI

You can now use the new UI to set up ValetudoTimers.

![image](https://user-images.githubusercontent.com/974410/133927153-e109c4a5-2bff-45e7-af07-2ae7a4eb9ace.png)

![image](https://user-images.githubusercontent.com/974410/133927151-76b2daa1-9e26-4b6d-a2e1-ba20c97e9829.png)

Note that timers are always stored and evaluated as UTC. The local time display is simply a convenience feature.

## MQTT config UI

MQTT configuration can now also be done via the new UI.

![image](https://user-images.githubusercontent.com/974410/133927206-8a4188a8-d178-46fb-9c2d-32907322f5f9.png)

It features a nice topic preview to make it easier for newcomers to interact with Valetudo via MQTT.

![image](https://user-images.githubusercontent.com/974410/133927215-e359d45a-c76e-47bb-a178-d1118cca25bc.png)

Also, the API will no longer return any MQTT credentials but instead replace them with `<redacted>` for security reasons.

## Logviewer UI

A basic log viewer has been added to the new UI. While still WIP, it already allows for live updates as well as filtering, which is a neat improvement.

![image](https://user-images.githubusercontent.com/974410/133927245-d4f98b09-0cc4-448d-9752-f939987f9e3c.png)

## Auto-empty Dock related features

If you own a Dreame Z10, you're now able to enable/disable the automatic dust collection via the Swagger UI.
Furthermore, you can also trigger it manually via either the UI, the REST API or MQTT, meaning that you can now fully customize the collection interval by using e.g. Home Assistant automations.

### Obstacle Avoidance

You can now disable the Obstacle Avoidance Feature of your Dreame L10/Z10 via the Swagger UI if you experience issues such as the robot refusing to drive onto a carpet.

### Consumables UI

The new UI now also features a consumables section with a neat hover feature hopefully helping newcomers better understand the consumables.

![image](https://user-images.githubusercontent.com/974410/133927281-cd81083a-be3f-415f-bf78-81818a0c26af.png)

### Runtime Information UI

The new UI about section has been extended to display data returned by the runtime information endpoint.

![image](https://user-images.githubusercontent.com/974410/133927292-91c520e9-210e-40c4-89a1-3c0fce037ad9.png)

### Features

* **core:** AutoEmptyDockAutoEmptyControlCapability ([eeeb46c](https://github.com/Hypfer/Valetudo/commit/eeeb46c8a2152049392b7df06c952dc44b101d98))
* **core:** AutoEmptyDockManualTriggerCapability ([5678d4f](https://github.com/Hypfer/Valetudo/commit/5678d4fd8e3dab9364182fa8f40fa09465bc77eb))
* **core:** ObstacleAvoidanceControlCapability ([e5fab92](https://github.com/Hypfer/Valetudo/commit/e5fab92172dd6c024201218bbe4889a770c3f7c2))
* **core:** Robot properties ([b1713df](https://github.com/Hypfer/Valetudo/commit/b1713dff26a2bcfd707bd88fa0dcf3acbde4faa1))
* **MockRobot:** MockAutoEmptyDockManualTriggerCapability ([31cb972](https://github.com/Hypfer/Valetudo/commit/31cb972d89cba474c3c8732e969015a5bb5d9851))
* **ui:** Add button to trigger dock auto empty manually ([#1087](https://github.com/Hypfer/Valetudo/issues/1087)) ([fa2def7](https://github.com/Hypfer/Valetudo/commit/fa2def7c65a2641ec6321fcdfb4c3975e607867d))
* **ui:** Consumables in new UI ([#1093](https://github.com/Hypfer/Valetudo/issues/1093)) ([144d6ed](https://github.com/Hypfer/Valetudo/commit/144d6edacdf264b5a835ee184c1639f982231d25))
* **ui:** Implement MQTT config ui ([#1080](https://github.com/Hypfer/Valetudo/issues/1080)) ([2dd83e3](https://github.com/Hypfer/Valetudo/commit/2dd83e3a8c8e8c1629f873b9d20a52f7337fbcd9))
* **ui:** Implement timer ui ([#1077](https://github.com/Hypfer/Valetudo/issues/1077)) ([abd6345](https://github.com/Hypfer/Valetudo/commit/abd63454588e510d23d55c4f08a1caad1cc7c649))
* **ui:** Log with SSE and display log in new UI ([#1092](https://github.com/Hypfer/Valetudo/issues/1092)) ([1721df7](https://github.com/Hypfer/Valetudo/commit/1721df7dd6969fba8acd63d726a8ca14d1977afd))
* **ui:** Navigation and Events in UI ([#1090](https://github.com/Hypfer/Valetudo/issues/1090)) ([27fd2ba](https://github.com/Hypfer/Valetudo/commit/27fd2ba97e98f534c20ca6e22b0d46510f355706))
* **ui:** Show MQTT config defaults ([711c312](https://github.com/Hypfer/Valetudo/commit/711c312cc41fff57867e5b97d4e867b3ef41b4c3))
* **ui:** Show MQTT topic preview ([#1082](https://github.com/Hypfer/Valetudo/issues/1082)) ([865905e](https://github.com/Hypfer/Valetudo/commit/865905ea58bfff9187c7463fce3a312b738a1e90))
* **ui:** Show runtime info on about page ([#1089](https://github.com/Hypfer/Valetudo/issues/1089)) ([ece2dc9](https://github.com/Hypfer/Valetudo/commit/ece2dc9d6d5baafa4d0aa291565087bf2e1f6148))
* **vendor.dreame:** DreameAutoEmptyDockAutoEmptyControlCapability ([a1fa6d8](https://github.com/Hypfer/Valetudo/commit/a1fa6d8a14bad572aca4ce6af714cacb9d4e1760))
* **vendor.dreame:** DreameAutoEmptyDockManualTriggerCapability ([b7aed1a](https://github.com/Hypfer/Valetudo/commit/b7aed1afa217ebd90f411c91844a28f05946460d))
* **vendor.dreame:** DreameDoNotDisturbCapability ([60972ca](https://github.com/Hypfer/Valetudo/commit/60972ca6dd8d8200d8c25a2e4e084c04f97ef8ce))
* **vendor.dreame:** DreameObstacleAvoidanceControlCapability ([ca9a499](https://github.com/Hypfer/Valetudo/commit/ca9a49939d9499dddfc9f77aede70c655758031a))
* **webserver:** Provide mqtt config properties ([7e6962b](https://github.com/Hypfer/Valetudo/commit/7e6962bd8094120168c960d550027efbf3a53d73))

### Bug Fixes

* **core:** Accepting/rejecting a pending map change should always invalidate a pending event ([94c8da2](https://github.com/Hypfer/Valetudo/commit/94c8da29c6f3f3adc3e9f02b546dd258e4bd30bc))
* **timers:** Prevent timer drift ([#1078](https://github.com/Hypfer/Valetudo/issues/1078)) ([105d02a](https://github.com/Hypfer/Valetudo/commit/105d02a67d502ba727e90251f2b9325837833ebd))
* **ui:** Improve word breaks in MQTT topic preview ([#1088](https://github.com/Hypfer/Valetudo/issues/1088)) ([8736f7f](https://github.com/Hypfer/Valetudo/commit/8736f7fc9ee1d76dcb8a5c21975f85040fbfd723))
* **ui:** Make timers use correct time zone offset ([#1079](https://github.com/Hypfer/Valetudo/issues/1079)) ([f790db3](https://github.com/Hypfer/Valetudo/commit/f790db3ec1eadcaa91726a75ac994b5d527058f4))
* **ui:** Remove staleTime: Infinity for queries that should actually refresh automatically ([b53d428](https://github.com/Hypfer/Valetudo/commit/b53d428846f3013768827296b193ed68f4ca0255))
* **ui:** Show old ui timer page even when there's no DoNotDisturbCapability ([ac19d5d](https://github.com/Hypfer/Valetudo/commit/ac19d5d2c1751c12d614ebf2c8cb701cdc7a0473))
* **ui:** Timer action controls Memo dependency ([#1091](https://github.com/Hypfer/Valetudo/issues/1091)) ([7ebbafc](https://github.com/Hypfer/Valetudo/commit/7ebbafce307fc5b3e07f73f3652d154a63d3a248))
* **ValetudoEvents:** Resetting a consumable should also set its event to processed ([d705c5b](https://github.com/Hypfer/Valetudo/commit/d705c5bc67cff5cb4b085893f428bf7794a26bb1)), closes [#1096](https://github.com/Hypfer/Valetudo/issues/1096)
* **webserver:** Remove obsolete mqtt config password stripping ([c4f6fab](https://github.com/Hypfer/Valetudo/commit/c4f6fabc62c69b6573782e31511f8df02ea2e7a3))
* **webserver:** Strip credentials from mqtt config REST response and remove ssh key upload ([5b66887](https://github.com/Hypfer/Valetudo/commit/5b6688778bd3efb884e027b51d2989e28c7252af))
* **webserver:** Terminate SSE connections on shutdown for a clean shutdown ([bfbf4dd](https://github.com/Hypfer/Valetudo/commit/bfbf4dd7192a443ca7e49d9efa0f382040370741))

<hr /><em>This discussion was created from the release <a href='https://github.com/Hypfer/Valetudo/releases/tag/2021.09.1'>Valetudo 2021.09.1</a>.</em>

**Comment by lindsaymarkward** [comment]
Thank you @Hypfer and @ccoors - we appreciate your work :)
Am I right that each time I want to upgrade my Roborock Gen2 S5, I follow the install procedure through dustbuilder?

**Reply by Hypfer** [reply]
https://valetudo.cloud/pages/general/upgrading.html
